rrocker-lib/protos/api.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+package rrocker.api;
+
+//Not the greatest design in the world since it doesn't encode
+//exactly which errors can be returned from which function
+//but I'm cutting this corner for simplicity's sake
+enum APIError {

```

Comment:
I wanted to mention incase you aren't aware, but gRPC does have it's own error mechanism and a number of predefined codes/results that are outside the message definition. I believe in tonic the codes are covered here: https://docs.rs/tonic/0.5.0/tonic/enum.Code.html

Reply:
You're right, I somehow missed that entirely. 
That'll make things quite a bit cleaner, I'll fix it right away.